src/Moryx.Products.Management/Implementation/Storage/ProductStorage.cs

```diff
@@ -123,9 +135,11 @@ public void Start()
             {
                 var strategy = StrategyFactory.CreateRecipeStrategy(config);
                 RecipeStrategies[config.TargetType] = strategy;
-
                 RecipeConstructors[config.TargetType] = ReflectionTool.ConstructorDelegate<IProductRecipe>(strategy.TargetType);
             }
+
+            _productTypes = TypeStrategies.Select(ts => ts.Value.TargetType).ToList();
```

Comment:
Almost what I meant. This still keeps two separate lists at runtime. If we ever support modification of strategies without restart, we run into the same issue. I still suggest moving the query in the getter. :-)